test_compileall2.py

```diff
@@ -955,83 +1033,159 @@ def test_ignore_symlink_destination(self):
         path = os.path.join(self.directory, "test", "same_some")
         os.makedirs(path)
 
         simple_script = script_helper.make_script(path, "test_same_bytecode", "a = 0")
         pyc_opt0 = importlib.util.cache_from_source(simple_script)
-        pyc_opt2 = importlib.util.cache_from_source(simple_script, optimization=2)
+        pyc_opt2 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(2)
+        )
 
         self.assertRunOK(path, "-q", "-o 0", "-o 2", "--hardlink-dupes")
-        compileall.compile_dir(path, quiet=True, optimize=[0, 2], hardlink_dupes=True)
-        self.assertTrue(os.stat(pyc_opt0).st_ino ==  os.stat(pyc_opt2).st_ino)
+
+        # Both files should have the same inode (hardlink)
+        self.assertEqual(os.stat(pyc_opt0).st_ino,  os.stat(pyc_opt2).st_ino)
 
         for pyc_file in pyc_opt0, pyc_opt2:
-            os.unlink(pyc_file)
+            try:
+                os.unlink(pyc_file)
+            except OSError:
+                # Python <= 3.4 does not create all the files
+                pass
 
         self.assertRunOK(path, "-q", "-o 0", "-o 2")
-        self.assertTrue(os.stat(pyc_opt0).st_ino !=  os.stat(pyc_opt2).st_ino)
+        
+        # Deduplication disabled, both pyc files should have different inodes
+        self.assertNotEqual(os.stat(pyc_opt0).st_ino, os.stat(pyc_opt2).st_ino)
 
     def test_hardlink_deduplication_different_bytecode_all_opt(self):
-        """Different bytecode produced for all levels of optimization"""
+        # "'''string'''\nassert 1" produces a different bytecode for all optimization levels
         path = os.path.join(self.directory, "test", "different_all")
         os.makedirs(path)
 
         simple_script = script_helper.make_script(path, "test_different_bytecode", "'''string'''\nassert 1")
         pyc_opt0 = importlib.util.cache_from_source(simple_script)
-        pyc_opt1 = importlib.util.cache_from_source(simple_script, optimization=1)
-        pyc_opt2 = importlib.util.cache_from_source(simple_script, optimization=2)
+        pyc_opt1 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(1)
+        )
+        pyc_opt2 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(2)
+        )
 
         self.assertRunOK(path, "-q", "-o 0", "-o 1", "-o 2", "--hardlink-dupes")
-        self.assertTrue(os.stat(pyc_opt0).st_ino != os.stat(pyc_opt1).st_ino != os.stat(pyc_opt2).st_ino)
+
+        # No hardlinks, bytecodes are different
+        self.assertNotEqual(os.stat(pyc_opt0).st_ino, os.stat(pyc_opt1).st_ino)
+        if compileall.PY35:
+            # Python 3.4 produces the same file for opt1 and opt2
+            self.assertNotEqual(os.stat(pyc_opt1).st_ino, os.stat(pyc_opt2).st_ino)
 
         for pyc_file in pyc_opt0, pyc_opt1, pyc_opt2:
```

Comment:
suggestion (not a strong one): if you use set() here, the try-pass won't be needed, if pyc_opt1 == pyc_opt2

Reply:
Good point, whatever makes the tests shorter :+1: Implemented